### ISSUE TITLE
Parser raise error when parse LiteralBlock

### ISSUE THREAD

**Issue by eonie** [bytesparadise/libasciidoc#474]
Here is the example
```
.....
List of Terms
-------------
var tmplStr = `<span>Example: {{.Value}}{{.Attributes.GetAsString "suffix"}}</span>`
var t = template.New("example")
var tmpl = template.Must(t.Parse(tmplStr))

output := &strings.Builder{}
content := strings.NewReader(`example::hello world[suffix=!!!!!]`)
libasciidoc.ConvertToHTML(context.Background(), content, output, renderer.DefineMacro(tmpl.Name(), tmpl))
.....
```
If dot number more than 4, the parser will raise an error.

```
Error: error while parsing the document: README.adoc:82:0 (4207): no match found, expected: !"\n", " ", ".", ":", "\r", "\r\n", "\t" or [\pL0-9]
Usage:
  libasciidoc [flags] FILE
  libasciidoc [command]

Available Commands:
  help        Help about the command
  version     Print the version and build info

Flags:
  -h, --help               help for libasciidoc
      --log string         log level to set [debug|info|warning|error|fatal|panic] (default "warning")
  -s, --no-header-footer   do not render header/footer (default: false)
  -o, --out-file string    output file (default: based on path of input file); use - to output to STDOUT

Use "libasciidoc [command] --help" for more information about a command.

error while parsing the document: README.adoc:82:0 (4207): no match found, expected: !"\n", " ", ".", ":", "\r", "\r\n", "\t" or [\pL0-9]
```

**Comment by xcoulon** [comment]
that's right, the delimiter should have 4 characters only. I do not check for now that the opening and closing delimiters have the same length. Is it a blocker for you, or can you just use `....` for now?

**Comment by eonie** [comment]
No, it is not a block for me, you can close it or keep it open until fixed.

**Comment by eonie** [comment]
I have a advice for you, maybe someday you can refact parser's code , it is a little hard to read and maintain.

**Comment by xcoulon** [comment]
> Would you like to refact parser's code someday in the future , it is a little hard to read and maintain.

what do you mean by "parser's code"? are you talking about https://github.com/bytesparadise/libasciidoc/blob/master/pkg/parser/parser.go or https://github.com/bytesparadise/libasciidoc/blob/master/pkg/parser/parser.peg? The former is generated with https://github.com/mna/pigeon, so there's little I can do about it ;)

**Comment by eonie** [comment]
I see , I was talking about `parser.go`, it is hard to read. I will take a look of `parser.peg`.

**Comment by xcoulon** [comment]
> I see , I was talking about `parser.go`, it is hard to read. I will take a look at `parser.peg`.

I agree with you. Hopefully, `parger.peg` is easier to understand ;)

**Comment by eonie** [comment]
> > I see , I was talking about `parser.go`, it is hard to read. I will take a look at `parser.peg`.
> 
> I agree with you. Hopefully, `parger.peg` is easier to understand ;)

Do you have references about PEG，I'm a beginner.

**Comment by xcoulon** [comment]
> > > I see , I was talking about `parser.go`, it is hard to read. I will take a look at `parser.peg`.
> > 
> > 
> > I agree with you. Hopefully, `parger.peg` is easier to understand ;)
> 
> Do you have references about PEG，I'm a beginner.

I learnt from the examples in https://github.com/mna/pigeon/tree/master/examples, to be honest.

**Comment by eonie** [comment]
I found that change `LiteralBlockDelimiter <- "...."` to `LiteralBlockDelimiter <- "...." "."*` could support more than 4 dot, is there any other problem:

**Comment by xcoulon** [comment]
> I found that change `LiteralBlockDelimiter <- "...."` to `LiteralBlockDelimiter <- "...." "."*` could support more than 4 dot, is there any other problem:

Yes, but then you should check that you have the same number of dots in the opening and closing delimiters. 
For example:
```
.....
inside the bloc
....

```
will not render properly:
```
<div class="literalblock">
<div class="content">
<pre>inside
....</pre>
</div>
</div>
```

Such a verification is what makes the parsing more complicated.

**Comment by eonie** [comment]
I test the sample, could render correct, no need to verify the opening and closing is match or not.
```
.........
Test LiteralBlockDelimiter dots more than 4 and not match with date
....
```
Here is the html result.
```
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<!--[if IE]><meta http-equiv="X-UA-Compatible" content="IE=edge"><![endif]-->
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="libasciidoc">
<title></title>
</head>
<body class="article">
<div id="header">
<h1></h1>
</div>
<div id="content">
<div class="literalblock">
<div class="content">
<pre>Test LiteralBlockDelimiter dots more than 4 and not match with date</pre>
</div>
</div>
</div>
<div id="footer">
<div id="footer-text">
Last updated 2020-01-14 22:02:53 +0800
</div>
</div>
</body>
</html>
```

**Comment by eonie** [comment]
Could I send a PR?

**Comment by xcoulon** [comment]
@eonie sorry, I'm afraid I wasn't clear in my previous comment: Asciidoctor requires that both opening and closing delimiters have the same length, so we should do the same here. Which is going to be a little more difficult.

**Comment by eonie** [comment]
Aha，I see. You are right, Asciidoctor could handle format like this
```
........
....
A test.
....
........
```
it will be parsed to
```
....
A test
....
```
it's more difficult for me, so I will just leave it,。
Hope someone can solve this

**Comment by xcoulon** [comment]
yes, that's another example of case where delimiter size matters, indeed.

**Comment by xcoulon** [comment]
fixed in #946
Feel free to reopen if you find a case that was not covered by the aforementioned PR ;)
